tracks/elixir/exercises/list-ops/mentoring.md

```diff
@@ -0,0 +1,136 @@
+### Reasonable solutions
+
+```elixir
+defmodule ListOps do
+  @spec count(list) :: non_neg_integer
+  def count(l) do
+    do_count(l, 0)
+  end
+
+  defp do_count([], total) do
+    total
+  end
+
+  defp do_count([_h|t], total) do
+    do_count(t, total + 1)
+  end
+
+  @spec reverse(list) :: list
+  def reverse(l) do
+    do_reverse(l, [])
+  end
+
+  defp do_reverse([h|[]], []) do
+    h
+  end
```

Comment:
This clause should be removed. It will cause a bug (`ListOps.reverse([1]) == 1`) and even if implemented correctly is unnecessary.